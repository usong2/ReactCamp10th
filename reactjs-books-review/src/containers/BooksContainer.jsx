import { connect } from "react-redux";
import Books from "../components/Books";
import axios from "axios";
import {
  startLoading,
  endLoading,
  setError,
  clearError,
  setBooks,
} from "../actions/index";

const mapStateToProps = (state) => ({
  books: state.books,
  loading: state.loading,
  error: state.error,
});

const mapDispatchProps = (dispatch) => ({
  requestBooks: (token) => {
    dispatch(startLoading());
    dispatch(clearError());
    axios
      .get("https://api.marktube.tv/v1/book", {
        headers: { Authorization: `Bearer ${token}` },
      })
      .then((res) => {
        dispatch(setBooks(res.data));
        dispatch(endLoading());
      })
      .catch((error) => {
        console.log(error);
        dispatch(setError(error));
        dispatch(endLoading());
      });
  },
});

export default connect(mapStateToProps, mapDispatchProps)(Books);
