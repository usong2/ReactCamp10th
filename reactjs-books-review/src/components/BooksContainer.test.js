import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureMockStore from "redux-mock-store";
import BooksContainer from "../containers/BooksContainer";

Enzyme.configure({ adapter: new Adapter() });

describe("BooksContainer", () => {
  it("renders properly", () => {
    const mockStore = configureMockStore();

    const store = mockStore({
      books: [],
      loading: false,
      error: null,
    });

    const component = mount(<BooksContainer store={store} />);
    expect(component).toMatchSnapshot();
  });
});
