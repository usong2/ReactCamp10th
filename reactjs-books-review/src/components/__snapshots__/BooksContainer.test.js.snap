// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksContainer renders properly 1`] = `
<Connect(Books)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Books
    books={Array []}
    error={null}
    loading={false}
    requestBooksPromise={[Function]}
    requestBooksSaga={[Function]}
    requestBooksThunk={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div />
  </Books>
</Connect(Books)>
`;
